pkg/ociclient/upload.go

```diff
@@ -0,0 +1,217 @@
+package ociclient
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	digest "github.com/opencontainers/go-digest"
+	specs "github.com/opencontainers/image-spec/specs-go"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const chunkSize = 100 * 1024 * 1024 // 100 MB chunks
```

Comment:
How about changing this to upload the whole file until it chunking works in the registry correctly?

Reply:
agreed, this only works for 100 MB files.  but larger gets really slow also.  
i'd prefer to keep this chunked and not working for > 100 MB than unknown size limitations.
product is not yet in prod